Feature/new demo site

eltonmesquita opened the pull request:
This PR restructure and implements a new layout for the demo website. You can see it [live here](https://animatecss-557d3.web.app/).

- [x] New layout  by @daneden 🎉🎉🎉
- [x] Merges the gh-pages branch with the current branch (dev), so everything can be automated
- [x] Animation list is automatically generated directly from the animate.css file
- [x] The Readme file was split for easy updates and gets compiled and injected in the template file (I've written a very simple static generator)
- [x] The demo website now holds the whole library documentation
- [x] The structure is easy to understand and maintain and there's just 1 dependency to make the maintenance easier
- [x] The docs still need to be rewritten and I'll make a PR soon defining a few things I think are important
- [x] Cross-browser testing

Solves #837 .

eltonmesquita commented:
Hey @WarenGonzaga , have a look at this PR. Your suggestions are welcome!

Any questions, feel free to ask 😉

WarenGonzaga commented:
Yow @eltonmesquita, I checked it and it looks awesome! However, there is something wrong with the scroll thing? it exceeds to my screen width...

![image](https://user-images.githubusercontent.com/15052701/80289403-bd493580-8770-11ea-84da-5be1f967fc98.png)

Also how about the color of the new website? Who choose the color? @daneden?

eltonmesquita commented:
@WarenGonzaga thanks! Yeah, Daniel made a rough sketch and I followed it respect the colour pallet and intentions.

Which browser, version, OS, etc that happens? I couldn't test it much yet but I did find some bugs related to grid on Mac's Chrome and had to give up of a few small features because of that.
It'd help a lot if you could test an provide feedback on as many up to date browsers as you can.

I've used lots of new CSS and js new stuff but in general it should work just fine in any modern browser. Feel free to give your opinions on the implementation too.

WarenGonzaga commented:
@eltonmesquita alright got it... I'm gonna test this again...

I'm using Windows 10 (latest version) with google chrome (latest version)

eltonmesquita commented:
I think that it was a small mistake on a `calc()` I just fixed it. I'll publish it tomorrow on the live demo. Thanks for the heads up!

WarenGonzaga commented:
@eltonmesquita sure bro... I might help in restructuring the documentation.

eltonmesquita commented:
I just updated the live demo with the fix. Can you check if it fix the issue for you @WarenGonzaga ?

eltonmesquita commented:
I've updated the PR with the rewritten documentation. For anyone reviewing this: I know that this is a massive PR, but keep your focus on the live version as it'll be quicker and easy.

On the documentation, I've tried to cover all the major issues that have being reported here through the years and give some orientation on the usage and how to make good use of the animations. I hope I've achieved my goals.

The new commits also solve #835 .

WarenGonzaga commented:
@eltonmesquita I'll take time to review this thank you!

WarenGonzaga commented:
Hi @eltonmesquita I ask my friends to review the demo site... they have a suggestion... I invited them to comment here.

ianvizarra commented:
Should it be like this..
<img width="1433" alt="Screen Shot 2020-05-01 at 11 59 10 PM" src="https://user-images.githubusercontent.com/2890710/80820025-b86d0180-8c08-11ea-86e4-f9cb08cb0eb8.png">

Rather than this?
<img width="1397" alt="Screen Shot 2020-05-01 at 11 55 24 PM" src="https://user-images.githubusercontent.com/2890710/80819927-8491dc00-8c08-11ea-8bd4-5d0b34d5c5e3.png">

I personally like the white background.

jkga commented:
The logo and the button are not centered
Dimension: `414x736`
![image](https://user-images.githubusercontent.com/10413754/80820492-8c05b500-8c09-11ea-80fe-e9208afd8c20.png)

WarenGonzaga commented:
@eltonmesquita do you have step to contribute to the demo site? So we can PR for proposed changes?

eltonmesquita commented:
Hi, thanks for the heads up. The intro part is not full-screen on purpose, so people can see that there's documentation right bellow that. It was something agreed previously with @daneden .

The docs/demo-site now reside in the `./docs` folder. To contribute with the documentation, just edit any md file in the `./docs/pages` folder. The template - which gets the MD files injected - resides on `./docs/template.html`. The CSS on `./doc/dist/style.css` and javascript on `./docs/dist/modules`.

There's no bundler or anything for CSS or Javascript, the compile step, `npm run docs`, just compile the Animate.css library to the `dist` folder, then compile and injects the MD files into the `template.html` file generating the `./docs/dist/index.html` file.

It's much less complicated then it sounds. Just have a look at the `./docs` folder and you'll get how it works!

WarenGonzaga commented:
> Hi, thanks for the heads up. The intro part is not full-screen on purpose, so people can see that there's documentation right bellow that. It was something agreed previously with @daneden .

Hi @eltonmesquita, how about adding "getting started" and it will smooth scroll to documentation? What about that?

eltonmesquita commented:
The problem of implementing the "Getting Started" button is that we'll 2 buttons on mobile, which we want to avoid. Do you see the way it is today as an issue? can you detail on that?

WarenGonzaga commented:
> The problem of implementing the "Getting Started" button is that we'll 2 buttons on mobile, which we want to avoid. Do you see the way it is today as an issue? can you detail on that?

Alright, it is not really an issue to me... that's only my suggestion from my perspective.

What do you think? @eltonmesquita

daneden commented:
Ohhh this is looking great so far! Thanks so much for working on it @eltonmesquita! I have a few pieces of design feedback:

- Could we use a more subtle/coordinated animation on initial load for the text? Maybe `zoomInDown` for both of them, perhaps with a _very_ slight delay between heading and subtitle?
- Now that I’m seeing it IRL, maybe the "above-the-fold" section should be `calc(100vh - 2.8rem)` (`2.8rem` is roughly the size of the "documentation" banner)
  - Also related, maybe it would be nice to add a little gradient fade-out to the bottom of the scrollable list of animations to make it more clear that the area is scrollable?
- A tooltip on the “copy” buttons would be nice! (e.g. "Copy class name to clipboard")
  - For accessibility reasons, I’d encourage making those actual `button` elements with the `title` attribute acting as the tooltip
- For code blocks, maybe reduce the relative font size (perhaps `.8em`) since monospace fonts appear larger
  - Also for code blocks, a slight border radius would be nice (probably `.25em` or so)

daneden commented:
> The problem of implementing the "Getting Started" button is that we'll 2 buttons on mobile, which we want to avoid. Do you see the way it is today as an issue? can you detail on that?

I don’t think two buttons on mobile is the end of the world FWIW :) and I like the suggestion of having it smooth scroll to docs @WarenGonzaga!

WarenGonzaga commented:
@daneden thank you! Also, I notice that the look and feel of the demo site is something similar to the branding of FedEx. Do you guys notice that? or only me?

eltonmesquita commented:
Cool, I'll start working on the feedbacks. @daneden , I've also rewritten the whole documentation and added some more sections, it's on the live demo already. Any feedback on it would be great!

daneden commented on docs/pages/00-intro.md:
```diff
@@ -0,0 +1,3 @@
+> Animate.css v4 brought some **breaking changes**, please refer to the [migration guide](#migration) before updating from v3.x and under.
+
+**Animate.css** is a bunch of cool, fun, and cross-browser animations for you to use in your projects. Great for emphasis, home pages, sliders, and general just-add-water-awesomeness.
```
```suggestion
**Animate.css** is a library of ready-to-use, cross-browser animations for use in your web projects. Great for emphasis, home pages, sliders, and attention-guiding hints.
```

daneden commented on docs/pages/01-usage.md:
````diff
@@ -0,0 +1,87 @@
+}
+```
+
+Be aware that some animations are dependent on the `animation-timing` property set on the animation's class. Changing, or not declaring it might harm the animation.
````
```suggestion
Be aware that some animations are dependent on the `animation-timing` property set on the animation's class. Changing or not declaring it might lead to unexpected results.
```

daneden commented on docs/pages/01-usage.md:
```diff
@@ -0,0 +1,87 @@
+
+#### CSS Custom Properties (CSS Variables)
+
+Since version 4, animate.css makes use of custom properties (also known as CSS variables) to define the animations duration, delay, and iteractions. This makes Animate.css very flexible and customizable. Need to change an animation duration? Just set a new value to globally or locally.
```
Nit: we should be consistent in the casing of Animate.css’ name—I think we should consistently Title-case it, but am open to suggestions (for example, it may be confusing that the _library_ is called `Animate.css`, but the file is called `animate.css`)

eltonmesquita replied:
Nice catch! That one went through! I prefer to Title-case it. Even though the file is not capitalized, it's an old convention to not capitalize files anyway. 🤷‍♂️

daneden commented on docs/pages/02-utilities.md:
````diff
@@ -0,0 +1,90 @@
+The `animate__animated` class has a default speed of `1s`. You can also customize the animations duration through the `--animate-duration` property, globaly or locally. This will affect both the animations and the utility classes. Example:
+
+```css
+/* All animations will take twice longer to finish */
````
```suggestion
/* All animations will take twice as long to finish */
```

daneden commented on docs/pages/02-utilities.md:
````diff
@@ -0,0 +1,90 @@
+}
+```
+
+Notice that some animations have a duration of less than 1 second. As we used the CSS `calc()` function, setting the duration through the `--animation-duration` property will respect these ratios. So, when you change the global duration all the animations will keep its consistency!
````
```suggestion
Notice that some animations have a duration of less than 1 second. As we used the CSS `calc()` function, setting the duration through the `--animation-duration` property will respect these ratios. So, when you change the global duration, all the animations will respond to that change!
```

daneden commented on docs/pages/03-best-practices.md:
```diff
@@ -0,0 +1,52 @@
+
+### Never animate root elements
+
+Animate the `<html/>` or `<body/>` tags is a no go as you'll just go through some browser weirdness.
```
This language is a bit vague; let’s see if we can smarten it up (linking to corroborating evidence can help).

daneden commented on docs/pages/03-best-practices.md:
```diff
@@ -0,0 +1,52 @@
+
+### Mind the initial and final state of your elements
+
+All the Animate.css animations include a CSS property called `animation-fill-mode` which controls the states of an element before and after animation. You can read more about it [here](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-fill-mode). Animate.css defaults to `animation-fill-mode: both`, but you can change it to suits your need.
```
```suggestion
All the Animate.css animations include a CSS property called `animation-fill-mode` which controls the states of an element before and after animation. You can read more about it [here](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-fill-mode). Animate.css defaults to `animation-fill-mode: both`, but you can change it to suit your needs.
```

daneden commented on docs/pages/07-custom-builds.md:
````diff
@@ -0,0 +1,53 @@
+```json
+/* on Animate.css package.json */
+"animateConfig": {
+  "prefix": "ani__"
````
```suggestion
  "prefix": "anim__"
```

eltonmesquita commented:
> * Could we use a more subtle/coordinated animation on initial load for the text? Maybe `zoomInDown` for both of them, perhaps with a _very_ slight delay between heading and subtitle?

It's actually totally random today. Every page reload is a new combination of entrance animations. Although I did like some of the random animations, sometimes it feels out of place. I did want some opinions on that! I think I'll switch to something more coordinated than.

eltonmesquita commented:
The last commit fixes:
[x] Use a more subtle/coordinated animation on initial load for the text
[x] "Above-the-fold" section should be calc(100vh - 2.8rem)
[ ] It would be nice to add a little gradient fade-out to the bottom of the scrollable list of animations to make it more clear that the area is scrollable. There's a browser bug - maybe specs quirks - that makes the element holding the gradient scroll with the list. I'm investigating it, but I think this is a non-blocker.
[x] A tooltip on the “copy” buttons.
[x] Make copy icons actual button elements with the title attribute acting as the tooltip. Done with a few adjustments.
[x] For code blocks reduce the relative font size
[x] For code blocks, adds a slight border radius

The [live demo](live here) is updated.

daneden commented:
> There's a browser bug - maybe specs quirks - that makes the element holding the gradient scroll with the list. I'm investigating it, but I think this is a non-blocker.

Ohhh interesting! Yeah I don’t think it’s a blocker, and the change to the section height actually seems to help. 

I have one last tiny request, which is to increase the padding and font size of the tooltips a little. They're a bit too tiny right now!

daneden commented on docs/dist/style.css:
```diff
@@ -0,0 +1,620 @@
+  position: absolute;
+  bottom: calc(100% + 0.56em);
+  right: -1em;
+  padding: 0.28em 0.5em;
```
```suggestion
  padding: 0.5em 0.75em;
```

daneden commented on docs/dist/style.css:
```diff
@@ -0,0 +1,620 @@
+  color: #fff;
+  border-radius: 4px;
+
+  font-size: 0.9em;
```
```suggestion

```

daneden commented:
This looks good to me 😃 awesome work! It’s yours to merge when you’re ready.

daneden commented:
Also yikes, that FOSSA check takes a while to run. I wonder if we want to remove it, or at least make it a non-blocking check.

eltonmesquita commented:
> Also yikes, that FOSSA check takes a while to run. I wonder if we want to remove it, or at least make it a non-blocking check.

Yeah, might be a good idea!
